pkg/logutil/logutil.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package logutil implements utilities for recording log output.
+package logutil
+
+import (
+	"io"
+	"os"
+
+	"github.com/nanmu42/limitio"
+)
+
+// NewWriterToFile creates a Writer that writes out output to a file path, creating necessary directories if needed, up to a maximum limit maxSize.
+func NewWriterToFile(maxSize int, dirPath, name string) (io.Writer, error) {
+	if err := os.MkdirAll(dirPath, 0700); err != nil {
```

Comment:
consider removing this logic, to create direction if it is missing, and ask that caller to make sure right directory is passed in.

One advantage is we not get in the case, that if an unintended directory is passed in, we don't end up create a new directory, and store logs there.

Reply:
good point, done.